src/OAuth2/Storage/Phalcon/Models/OauthUsers.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace OAuth2\Storage\Phalcon\Models;
+
+use Phalcon\Mvc\Model\Validator\Email as Email;
+
+class OauthUsers extends \Phalcon\Mvc\Model
+{
+
+    /**
+     *
+     * @var string
+     */
+    public $username;
+
+    /**
+     *
+     * @var string
+     */
+    public $password;
+
+    /**
+     *
+     * @var string
+     */
+    public $first_name;
+
+    /**
+     *
+     * @var string
+     */
+    public $last_name;
+
+    /**
+     *
+     * @var string
+     */
+    public $email;
+
+    /**
+     *
+     * @var integer
+     */
+    public $email_verified;
+
+    /**
+     *
+     * @var string
+     */
+    public $scope;
+
+    /**
+     * Allows to query a set of records that match the specified conditions
+     *
+     * @param mixed $parameters
+     * @return OauthUsers[]
+     */
+    public static function find($parameters = null)
+    {
+        return parent::find($parameters);
+    }
+
+    /**
+     * Allows to query the first record that match the specified conditions
+     *
+     * @param mixed $parameters
+     * @return OauthUsers
+     */
+    public static function findFirst($parameters = null)
+    {
+        return parent::findFirst($parameters);
+    }
+
+    /**
+     * Initialize method for model.
+     */
+    public function initialize()
+    {
+        $this->keepSnapshots(true);
+        $this->setSource("'oauth__users'");
+        $this->hasMany('username', 'OAuth2\Storage\Models\OauthAccessTokens', 'user_id', array("alias" => "AccessTokens"));
+        $this->hasMany('username', 'OAuth2\Storage\Models\OauthRefreshTokens', 'user_id', array("alias" => "RefreshTokens"));
+    }
+
+    /**
+     * Validations and business logic
+     *
+     * @return boolean
+     */
+    public function validation()
+    {
+        $this->validate(
+            new Email(
+                array(
+                    'field' => 'email',
+                    'required' => true,
+                )
+            )
+        );
+
+        if ($this->validationHasFailed() == true) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns table name mapped in the model.
+     *
+     * @return string
+     */
+    public function getSource()
+    {
+        return 'oauth__users';
+    }
+
+    /**
+     * @param mixed $parameters
+     * @return \OAuth2\Storage\Models\OauthAccessTokens[]
+     */
+    public function getAccessTokens($parameters = null)
+    {
+        return $this->getRelated('AccessTokens', $parameters);
+    }
+
+    /**
+     * @param mixed $parameters
+     * @return \OAuth2\Storage\Models\OauthRefreshTokens[]
+     */
+    public function getRefreshTokens($parameters = null)
+    {
+        return $this->getRelated('RefreshTokens', $parameters);
+    }
+}
```

Comment:
Ah, got it. Thanks for pointing them out. I'll clean it up ASAP. What do you
think about all the model files though? Too much for the style of the repo?  
  
Sent from [Nylas N1](https://link.nylas.com/link/1bn4fr7kwwyc20cix4nnlmvaw
/local-818fd84c-
52e8/0?redirect=https%3A%2F%2Fnylas.com%2Fn1%3Fref%3Dn1&r=cmVwbHkrMDAxYjM3OWI2YjFiNTYzOTQ0ZjljNDAyZGMyZTg3OTZkMWNmNzRmNjEzNDY5N2I4OTJjZjAwMDAwMDAxMTM4YzA3NWY5MmExNzBjZTA0MWQ2YWMzQHJlcGx5LmdpdGh1Yi5jb20=),
the extensible, open source mail client.  
![](https://link.nylas.com/open/1bn4fr7kwwyc20cix4nnlmvaw/local-818fd84c-
52e8?r=cmVwbHkrMDAxYjM3OWI2YjFiNTYzOTQ0ZjljNDAyZGMyZTg3OTZkMWNmNzRmNjEzNDY5N2I4OTJjZjAwMDAwMDAxMTM4YzA3NWY5MmExNzBjZTA0MWQ2YWMzQHJlcGx5LmdpdGh1Yi5jb20=)

On Jun 29 2016, at 6:02 pm, Brent Shaffer &lt;notifications@github.com&gt;
wrote:  

> In
[src/OAuth2/Storage/Phalcon/Models/OauthUsers.php](https://github.com/bshaffer/oauth2
-server-
php/pull/743#discussion_r69036739&r=cmVwbHkrMDAxYjM3OWI2YjFiNTYzOTQ0ZjljNDAyZGMyZTg3OTZkMWNmNzRmNjEzNDY5N2I4OTJjZjAwMDAwMDAxMTM4YzA3NWY5MmExNzBjZTA0MWQ2YWMzQHJlcGx5LmdpdGh1Yi5jb20=):

>  
>  
>     > +     * @return \OAuth2\Storage\Models\OauthAccessTokens[]

>     > +     */

>     > +    public function getAccessTokens($parameters = null)

>     > +    {

>     > +        return $this->getRelated('AccessTokens', $parameters);

>     > +    }

>     > +

>     > +    /**

>     > +     * @param mixed $parameters

>     > +     * @return \OAuth2\Storage\Models\OauthRefreshTokens[]

>     > +     */

>     > +    public function getRefreshTokens($parameters = null)

>     > +    {

>     > +        return $this->getRelated('RefreshTokens', $parameters);

>     > +    }

>     > +}

>  
>

> add newline :P

>

> —  
> You are receiving this because you authored the thread.  
> Reply to this email directly, [view it on
GitHub](https://github.com/bshaffer/oauth2-server-
php/pull/743/files/eedce8fc62e86eb3622d1e4b788470d7319d4128#r69036739&r=cmVwbHkrMDAxYjM3OWI2YjFiNTYzOTQ0ZjljNDAyZGMyZTg3OTZkMWNmNzRmNjEzNDY5N2I4OTJjZjAwMDAwMDAxMTM4YzA3NWY5MmExNzBjZTA0MWQ2YWMzQHJlcGx5LmdpdGh1Yi5jb20=),
or [mute the
thread](https://github.com/notifications/unsubscribe/ABs3m1yn9fmL2U6MObO_J8nn4DMZdWETks5qQutfgaJpZM4JBXTs&r=cmVwbHkrMDAxYjM3OWI2YjFiNTYzOTQ0ZjljNDAyZGMyZTg3OTZkMWNmNzRmNjEzNDY5N2I4OTJjZjAwMDAwMDAxMTM4YzA3NWY5MmExNzBjZTA0MWQ2YWMzQHJlcGx5LmdpdGh1Yi5jb20=).![](https://github.com/notifications/beacon/ABs3mwK3sUwD1XuCnvtrPznDU7YI1Zzvks5qQutfgaJpZM4JBXTs.gif)